batch/batch/aioclient.py

```diff
@@ -105,11 +106,21 @@ def __init__(self, client, id, attributes):
 
 
 class BatchClient:
-    def __init__(self, session, url=None):
+    def __init__(self, session, url=None, token_file=None, token=None, headers=None):
         if not url:
             url = 'http://batch.default'
         self.url = url
         self._session = session
+        if token is None:
+            token_file = (token_file or
+                          os.environ.get('HAIL_TOKEN_FILE') or
+                          os.path.expanduser('~/.hail/token'))
+            if os.path.exists(token_file):
+                with open(token_file) as f:
+                    token = f.read()
+        if token is not None:
```

Comment:
Assert token is not none

Reply:
made it into a ValueError if there is no token